### PR TITLE
fix(web): cache static assets

### LINKED ISSUE #1894: `kimi web` serves static asset files inefficiently without caching
### What version of Kimi Code is running?

0.27.0

### Which open platform/subscription were you using?

platform.kimi.ai

### Which model were you using?

_No response_

### What platform is your computer?

_No response_

### What issue are you seeing?

When accessing `kimi web` from less powerful devices (e.g. server = Raspberry Pi, client = phone), the loading screen can take about 20 seconds to complete. While checking with DevTools, I notice that static assets aren't cached.

### What steps can reproduce the bug?

Run `kimi web`. Open Chrome DevTools. Observe "Network". Refresh kimi's web page. Check that static asserts files like `assets/index-Dy7xs5tu.js` are served repeatedly (HTTP 200) instead of serving from cache.

### What is the expected behavior?

Assets files should be cached by the browser and skipped serving.

### Additional information

Suggested fix: a few lines change.

Add `Cache-Control: public, max-age=31536000, immutable` to around here:

https://github.com/MoonshotAI/kimi-code/blob/ceb158dc54586f254819edbc83c27e21dca1ecf6/packages/kap-server/src/routes/webAssets.ts#L60

Note: the 20-second slowness turned out to be primarily caused by another issue, reported separated as #1904.

### PR DESCRIPTION
**Pull request by floze-the-genius**
## Related Issue

Resolves #1894

## Problem

`kimi web` returns HTTP 200 for fingerprinted Vite assets on every reload because the embedded server does not set a cache policy. This makes repeat page loads unnecessarily slow on lower-powered servers and clients.

## What changed

- Serve generated `/assets/` files with `Cache-Control: public, max-age=31536000, immutable`.
- Serve the application shell and SPA fallback with `Cache-Control: no-cache` so deployments are revalidated instead of pinning an old `index.html`.
- Add route-level regressions for both cache policies.
- Add a patch changeset for the bundled CLI web experience.

## Validation

- `pnpm exec vitest run packages/kap-server/test/webAssets.test.ts` (4 passed)
- `pnpm --filter @moonshot-ai/kap-server test` (675 passed)
- `pnpm --filter @moonshot-ai/kap-server run typecheck`
- `pnpm --filter @moonshot-ai/kap-server run build`
- `pnpm exec changeset status --output=.tmp-changeset-status.json`
- `git diff --check`

## AI assistance

AI-assisted tooling was used to help implement and test this focused fix. I reviewed the resulting diff, cache behavior, edge cases, and validation results before submission.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/MoonshotAI/kimi-code/blob/main/CONTRIBUTING.md) document.
- [x] I have linked a related issue, or explained the problem above.
- [x] I have added tests that prove the fix works.
- [x] Ran `gen-changesets` skill, or this PR needs no changeset.
- [x] Ran `gen-docs` skill, or this PR needs no doc update.

### CODE REVIEW

File: packages/kap-server/src/routes/webAssets.ts [hunk 1]
```diff
@@ -56,10 +56,17 @@ async function serveWebAsset(
 
   return reply
     .type(mimeType(filePath))
+    .header('Cache-Control', cacheControl(requestUrl.pathname))
```
[the remark this answers is not in the document]
Comment (floze-the-genius) [reply packages/kap-server/src/routes/webAssets.ts]: Fixed in ca383b11. Cache policy is now derived from the resolved file path, so SPA fallbacks such as `/assets/preview` receive `no-cache` while actual files under the assets directory remain immutable. Added the fallback URL to the regression matrix; focused tests (5/5), typecheck, and build pass.

**Comment by floze-the-genius** [comment]
@codex review

**Comment by puabiq** [comment]
@codex review
